jina/drivers/__init__.py

```diff
@@ -362,6 +364,49 @@ def _traverse_rec(self, docs, parent_doc, parent_edge_type, path, *args, **kwarg
             self._apply_all(docs, parent_doc, parent_edge_type, *args, **kwargs)
 
 
+class SetRecursiveMixin:
+    """
+     The full datastructure version of :class:`RecursiveMixin`, to be mixed in with :class:`BaseRecursiveDriver`
```

Comment:
this one is not to be used in Index and EncodeDrivrr right? Why is this change applied directly on RecursiveMixin (and maybe renaming it?)

Reply:
Because I want to refactor it piece by piece. Finally, the `RecursiveMixin` should be deleted.